ada_feeding_actionselect/ada_feeding_actionselect/policy_service.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+"""
+This module defines a node that launches a 2 ROS2 services.
+This service implement AcquisitionSelect and AcquisitionReport.
+"""
+
+# Standard imports
+from typing import Dict, Union
+import numpy as np
+
+# Third-party imports
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import CompressedImage, Image
+from rcl_interfaces.msg import ParameterDescriptor, ParameterType
+
+# Internal imports
+from ada_feeding.helpers import import_from_string
+from ada_feeding_perception.helpers import get_img_msg_type, ros_msg_to_cv2_image
+from ada_feeding_actionselect.policies import Policy
+from ada_feeding_actionselect.adapters import ContextAdapter, PosthocAdapter
+from ada_feeding_msgs.srv import AcquisitionSelect, AcquisitionReport
+
+
+class PolicyServices(Node):
+    """
+    The PolicyServices node initializes a 3 things:
+    a context adapter converts the incoming mask and images to a context vector
+    a policy recommends an action given the context
+    a posthoc adapter converts any incoming posthoc data to a posthoc vector.
+
+    This node collects images (rgb + depth) to pass to the context adapter
+    and creates services to implement AcquisitionSelect and AcquisitionReport.
+
+    TODO: optionally record data (context + posthoc + loss) based on param
+    """
+
+    def __init__(self):
+        """
+        Declare ROS2 Parameters
+        """
+        super().__init__("policy_service")
+
+        # Name of the Policy
+        policy_param = self.declare_parameter(
+            "policy",
+            descriptor=ParameterDescriptor(
+                name="policy",
+                type=ParameterType.PARAMETER_STRING,
+                description=(
+                    "Which policy to use. "
+                    "Names must correspond to their own param namespace within this node."
+                ),
+                read_only=True,
+            ),
+        )
+        policy_name = policy_param.value
+        if policy_name is None:
+            raise ValueError("Policy Name is required.")
+
+        # Import the policy class
+        policy_cls_param = self.declare_parameter(
+            f"{policy_name}.policy_class",
+            descriptor=ParameterDescriptor(
+                name="policy_class",
+                type=ParameterType.PARAMETER_STRING,
+                description=(
+                    "The class of the policy to run, must subclass "
+                    "Policy. E.g., ada_feeding_actionselect.policies.ConstantPolicy"
+                ),
+                read_only=True,
+            ),
+        )
+
+        policy_cls_name = policy_cls_param.value
+        if policy_cls_name is None:
+            raise ValueError("Policy Class is required.")
+
+        policy_cls = import_from_string(policy_cls_name)
+        assert issubclass(policy_cls, Policy), f"{policy_cls_name} must subclass Policy"
+
+        kwargs = self.get_kwargs(f"{policy_name}.kws", f"{policy_name}.kwargs")
```

Comment:
Nit: can you rename this to `policy_kwargs` (and make the corresponding change when you initialize `self.policy` below?) `kwargs` is typically used for keyword arguments passed into a function e.g., `def __init__(self, **kwargs)`